subsys/nrf_security/src/drivers/cracen/cracenpsa/src/ed25519.c

```diff
@@ -0,0 +1,389 @@
+	if (status != 0) {
+		return status;
+	}
+	status = sx_hash_feed(&hashopctx, workmem + SX_ED25519_SZ, SX_ED25519_SZ);
+	if (status != 0) {
+		return status;
+	}
+	status = sx_hash_feed(&hashopctx, message, message_length);
+	if (status != 0) {
+		return status;
+	}
+	status = sx_hash_digest(&hashopctx, workmem + SX_ED25519_SZ);
+	if (status != 0) {
+		return status;
+	}
+	status = sx_hash_wait(&hashopctx);
+	if (status != 0) {
+		return status;
+	}
+	return status;
+}
+
+int ed25519_sign_internal(const uint8_t *priv_key, char *signature, const uint8_t *message,
+			size_t message_length, int prehash)
+{
+	char workmem[5 * SX_ED25519_SZ];
+	struct sxhash hashopctx;
+	char pointr_buffer[SX_ED25519_DGST_SZ];
+	char *pointr = pointr_buffer;
+	int status;
+
+	/*Hash the private key, the digest is stored in the first 64 bytes of workmem*/
+	status = ed25519_hash_privkey(hashopctx, workmem, priv_key);
+	if (status != 0) {
+		return status;
+	}
+
+	/* Obtain r by hashing (prefix || message), where prefix is the second
+	 * half of the private key digest.
+	 */
+	status = ed25519_calculate_r(hashopctx, workmem, message, message_length, prehash);
+	/* Perform point multiplication R = [r]B. This is the encoded point R,
+	 * which is the first part of the signature.
+	 */
+	status = sx_ed25519_ptmult((const struct sx_ed25519_dgst *)(workmem + 3 * SX_ED25519_SZ),
+				(struct sx_ed25519_pt *)pointr);
+	if (status != 0) {
+		return status;
+
+	}
+	/* The secret scalar s is computed in place from the first half of the
+	 * private key digest.
+	 */
+	decode_scalar_25519(workmem);
+
+	/* Clear second half of private key digest: sx_async_ed25519_ptmult_go()
+	 * works on an input of SX_ED25519_DGST_SZ bytes.
+	 */
+	safe_memset(workmem + SX_ED25519_SZ, sizeof(workmem) - SX_ED25519_SZ, 0, SX_ED25519_SZ);
+	/* Perform point multiplication A = [s]B,
+	 * to obtain the public key A. which is stored in workmem[32:63]
+	 */
+	status = sx_ed25519_ptmult((const struct sx_ed25519_dgst *)(workmem),
+	(struct sx_ed25519_pt *)(struct sx_ed25519_pt *)((char *)workmem + SX_ED25519_PT_SZ));
+
+	if (status != 0) {
+		return status;
+	}
+
+	status = ed25519_calculate_k(hashopctx, workmem, pointr, message, message_length, prehash);
+	if (status != 0) {
+		return status;
+	}
+
+	/* Compute (r + k * s) mod L. This gives the second part of the
+	 * signature, which is the encoded S which is stored in pointr.
+	 */
+
+	status = sx_ed25519_sign(
+		(const struct sx_ed25519_dgst *)(workmem + SX_ED25519_SZ),
+		(const struct sx_ed25519_dgst *)(workmem + 3 * SX_ED25519_SZ),
+		(const struct sx_ed25519_v *)workmem,
+		(struct sx_ed25519_v *)(pointr + SX_ED25519_PT_SZ));
+	if (status != 0) {
+		return status;
+	}
+
+	memcpy(signature, pointr, SX_ED25519_DGST_SZ);
+
+	return status;
+
+}
+
+
+int ed25519_sign(const uint8_t *priv_key, char *signature,
+		const uint8_t *message, size_t message_length)
+{
+	return ed25519_sign_internal(priv_key, signature, message, message_length, 0);
```

Comment:
```suggestion
	return ed25519_sign_internal(priv_key, signature, message, message_length, false);
```
Same further down, booleans to say yes/no rather than integers.